lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -0,0 +1,174 @@
+
+
+class ClearTools(ToolCommand):
+    """Clear registered tools"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging(logfile=self.pbench_log)
+
+    def execute(self):
+        errors = 0
+
+        self.tool_group_dir = self.tool_group_dir(self.context.group)
+        if not self.tool_group_dir.exists():
+            return 1
+
+        try:
+            self.context.remote = self.context.remote.split(",")
+        except Exception:
+            # We were not given any remotes on the command line, build the list
+            # from the tools group directory.
+            self.context.remote = self.remote(self.tool_group_dir)
+
+        for remote in self.context.remote:
+            tg_dir_r = self.tool_group_dir / remote
+            if not tg_dir_r.exists():
+                self.logger.warn(
+                    f'The given remote host, "{remote}", is not a directory in {self.tool_group_dir}.'
+                )
+                continue
+
+            if not self.context.name:
+                # Discovery all the tools registered for this remote
+                self.context.name = self.tools(tg_dir_r)
+
+            if self.context.name:
+                for name in self.context.name:
+                    status = self._clear_tools(name, remote)
+                    if status != 0:
+                        errors += 1
+
+            tool_files = [p.name for p in tg_dir_r.iterdir()]
+
+            if "__label__" in tool_files:
+                label = tg_dir_r / "__label__"
+                try:
+                    os.unlink(label)
+                except OSError:
+                    self.logger.error(f"Failed to remove label for remote {tg_dir_r}")
+                    errors += 1
+
+            if self.is_empty(tg_dir_r):
+                self.logger.info(f'All tools removed from host, "{tg_dir_r.name}"')
+                try:
+                    shutil.rmtree(tg_dir_r)
+                except OSError:
+                    self.logger.error(f"Failed to remove remote directory {tg_dir_r}")
+                    errors += 1
+
+        return errors
+
+    def _clear_tools(self, name, remote):
+        """Remove specified tool and associated files"""
+        tpath = self.tool_group_dir / remote / name
+        try:
+            if tpath.exists():
+                os.unlink(tpath)
+                noinstall = pathlib.Path("f{tpath}.__noinstall__")
+                if noinstall.exists():
+                    os.unlink(noinstall)
+
+                self.logger.info(
+                    f'Removed "{name}" from host, "{remote}", in tools group, "{self.context.group}"'
+                )
+                return 0
+        except Exception as e:
+            self.logger.error(f"Failed to remove {tpath}: {e}")
+            return 1
+
+    def is_empty(self, path):
+        """Determine if directory is empty or not"""
+        return bool(not any([p for p in path.iterdir()]))
```

Comment:
Seems like we don't need to build an entire list first via a list comprehension, but would want to exit on the first directory entry found, something like:

```
def is_empty(self, path):
    for dirent in path.iterdir():
        return False
    return True
```